src/main/java/com/cloudbees/groovy/cps/impl/FunctionCallBlock.java

```diff
@@ -97,7 +100,11 @@ private Next dispatchOrArg() {
                     return k.receive(v);
                 } else {
                     // regular method call
-                    return methodCall(e,loc,k,lhs,name,args);
+                    if (safe && lhs == null) {
+                        return k.receive(null);
```

Comment:
Check this late implies that `x?.foo(a())` would evaluate `a()`. Is that what Groovy actually does? I think it might skip evaluating arguments altogether if `lhs` evaluates to `null`.